storage-node/src/cache/data_store_cache/mod.rs

```diff
@@ -0,0 +1,43 @@
+use async_trait::async_trait;
+use bytes::Bytes;
+use tokio::sync::mpsc::Receiver;
+
+use crate::{error::ParpulseResult, storage_reader::StorageReaderStream};
+
+pub mod memdisk;
+
+#[async_trait]
+pub trait DataStoreCache {
+    async fn get_data_from_cache(
+        &mut self,
```

Comment:
I have to make these 2 methods mutable, since they will modify the mem_store, and which is not guarded with RwLock currently.
I can revert them back and add RwLock to mem_store, but I think it will violate the rule that let data_store handle its concurrency.
@xx01cyx What do you think?

Reply:
Feel free to modify it!